arlas-core/src/main/java/io/arlas/server/core/ElasticAdmin.java

```diff
@@ -43,13 +43,20 @@
     private static Logger LOGGER = LoggerFactory.getLogger(ElasticAdmin.class);
 
     public Client client;
+    private ColumnFilterChecker columnFilterChecker;
 
     public ElasticAdmin(Client client) {
         this.client = client;
+        this.columnFilterChecker = new ColumnFilterChecker();
     }
 
     @SuppressWarnings({"unchecked", "rawtypes"})
-    public CollectionReferenceDescription describeCollection(CollectionReference collectionReference) throws IOException {
+    public CollectionReferenceDescription describeCollection(CollectionReference collectionReference) {
+        return this.describeCollection(collectionReference, Optional.empty());
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public CollectionReferenceDescription describeCollection(CollectionReference collectionReference, Optional<Set<String>> columnFilterPredicates) {
```

Comment:
I think it will be better to pass the 2nd argument with the same type for `describeCollection` and `describeAllCollections`.
I see that you use `Optional<Set<String>> columnFilterPredicates` in this recursive function, so maybe we can add a new method : 
`public CollectionReferenceDescription describeCollection(CollectionReference collectionReference, Optional<String> columnFilter)` that fetches the predicate first and then calls this current method

Reply:
It is not possible the way you propose it, because if 2 methods have a signature with same objects but different generic class (ex: void toto(Optional<String>) vs titi(Optional<List<String>>) the compiler cannot distinguish them.
Using Optional<String> in both methods to have the same signature.